### PR TITLE
Add documenation to README.md and RELEASING.md

### PR DESCRIPTION
**Pull request by pichlermarc**
This PR adds
- a basic repository description to `README.md` 
- a description where to find the published releases to `README.md` 
- a description of the release process to `RELEASING.md`

### CODE REVIEW

**Comment by pichlermarc** [comment]
Please note that the failing PR build is unrelated as this PR only changes Markdown files.

**Review by jkwatson — APPROVED** [review]

**Comment by jkwatson** [comment]
something is definitely messed up with the PR build. It's trying to download an unreleased version of the proto zip....

File: README.md [hunk 1]
```diff
@@ -1,3 +1,37 @@
...
+<dependency>
+  <groupId>io.opentelemetry.proto</groupId>
+  <artifactId>opentelemetry-proto</artifactId>
+  <version>{{version}}</version>
+</dependency>
+```
+
+### Gradle
+
+```groovy
```
Comment (anuraaga) [inline README.md:31]: Let's use kotlin syntax (since it is also compatible with Groovy)

File: README.md [hunk 2]
```diff
@@ -1,3 +1,37 @@
...
+implementation 'io.opentelemetry.proto:opentelemetry-proto:{{version}}'
+```
```
Comment (anuraaga) [inline README.md:33]: Thanks for the help! Yeah the build fails until a new release of the proto has been made upstream, due to a shortcut to make the release as simple as possible. Will look into improving it

File: README.md (hunk 1 above)
Comment (pichlermarc) [reply README.md]: Good idea! Fixed in b1682b9.

File: README.md (hunk 2 above)
Comment (pichlermarc) [reply README.md:33]: Thank you for pointing this out. :)
I just wanted to mention it in this PR as it has no effect on the build. Should I create an issue for that in the meantime?

Comment (anuraaga) [reply README.md:33]: Sure :)

**Review by anuraaga — APPROVED** [review]

**Pull request merged**